portfolio/src/main/webapp/script.js

```diff
@@ -56,7 +56,17 @@ function openContact(){
 async function serverCommunication(){
   const serverResponse = await fetch("/data");
   console.log(serverResponse);
-  const responseToText = await serverResponse.text();
-  console.log(responseToText);
-  document.getElementById("greeting-container").innerHTML = responseToText;
+  const responseToJson = await serverResponse.json();
+  
+  //create object with response from server
+  myObject = responseToJson;
```

Comment:
Nit: may be good to use a slightly more concrete name for variables, even if this is Javascript. Consider something like "wordsArray", etc.